### PR TITLE
@kanaabe => CTA Email modal

### LINKED ISSUE #28: Mobile Email Signup Modal
This modal should display for users that have viewed 4 articles in the past 30 days. Closing the modal should hide it for the user for 30 days. T

It would be great if we could auto-populate the form for signed in / not signed up users. 

The form should not display for email traffic. 

For these we should be tracking: impressions, signups, and closes.

🌟 

screen shot 2016-09-19 at 1 35 57 pm

Sketch File: https://cl.ly/37263Y2P0l2x

### PR DESCRIPTION
**Pull request by eessex**
closes https://github.com/artsy/microgravity/issues/28
- Add cookies to track client article views
- Build CTA email popup to display in articles according to cookie status

### CODE REVIEW

**Comment by eessex** [comment]
Having trouble getting the shrinkwrap back to 0 after installing and removing referer-parser, i copied and pasted from the master but it still seems different -- let me know if you have suggestions for this issue.

**Comment by kanaabe** [comment]
Since we installed a new module, `npm-shrinkwrap.json` should have a diff! It will likely include a bunch of changes from other modules, which is fine since it means those other modules have been updated/had version bumps. We should do a `rm npm-shrinkwrap.json` and `npm shrinkwrap --dev` again to get the updates for `referer-parser`. 

Also, might be a good idea to mark this PR's title with [WIP] or [HOLD] since we're waiting for feedback from Owen about the popup for Eigen.

File: analytics/articles.js [hunk 1]
```diff
@@ -72,6 +72,14 @@ if(location.pathname.match('/article/')){
     });
   });
 
+  analyticsHooks.on('impressions:editorial-signup', function(options){
+    analytics.track('Viewed popup for editorial email', {
```
Comment (kanaabe) [inline analytics/articles.js]: The message should mimic these existing article impression analytics: https://github.com/artsy/microgravity/blob/master/analytics/article_impressions.js#L135

File: components/cta_bar/index.styl [hunk 2]
```diff
@@ -1,14 +1,69 @@
...
+        line-height 1.25
+    .cta-bar-defer a
+      font-size 22px
+      color #ccc
+      text-decoration none
+    .cta-bar-form-editorial
+      position relative
+      .bordered-input
+        border-width 1px
+        font-size 18px
```
Comment (kanaabe) [inline components/cta_bar/index.styl]: for font sizes, check out this style guide: https://www.artsy.net/style-guide/typography

File: components/cta_bar/index.styl [hunk 3]
```diff
@@ -1,14 +1,69 @@
...
+  display flex
+  align-items flex-end
+  min-height 35vh
+  .vam-outer
+    background-color white
+    color black
+    padding 10px 20px
+    margin 10px
+    width 100%
+    border 1px solid #E5E5E5
```
Comment (kanaabe) [inline components/cta_bar/index.styl]: we should use the colors from the style guide here as well! `gray-lighter-color`

File: components/cta_bar/index.styl [hunk 4]
```diff
@@ -1,14 +1,69 @@
 @import '../stylus_lib'
 
+.cta-bar.editorial-signup
+  padding 0
+  background none transparent
+  background linear-gradient(to bottom, rgba(0,0,0,0), rgba(0,0,0,.25))
+  display flex
+  align-items flex-end
+  min-height 35vh
```
Comment (kanaabe) [inline components/cta_bar/index.styl]: hmm what's this for? might be nicer to set a specific height instead from the Sketch file.

File: components/cta_bar/index.styl [hunk 5]
```diff
@@ -1,14 +1,69 @@
...
+    border 1px solid #E5E5E5
+    .cta-bar-header-editorial
+      margin 10px 0 15px 0
+      > h2
+        garamond-size('s-headline')
+        text-align left
+        line-height 1.25
+    .cta-bar-defer a
+      font-size 22px
+      color #ccc
```
Comment (kanaabe) [inline components/cta_bar/index.styl]: 🌈 `gray-color`

File: components/cta_bar/index.styl [hunk 6]
```diff
@@ -1,14 +1,69 @@
...
+        right 10px
+        avant-garde-size('s-headline', true)
+        line-height 1em
+      .subscribed
+        position absolute
+        top 0
+        left 0
+        right 0
+        bottom 0
+        background-color #ccc
```
Comment (kanaabe) [inline components/cta_bar/index.styl]: 🌈

File: components/cta_bar/view.coffee [hunk 7]
```diff
@@ -18,16 +18,17 @@ module.exports = class CTABarView extends Backbone.View
     name: 'cta_bar'
     mode: 'email'
     persist: true
+    expires: 31536000
```
Comment (kanaabe) [inline components/cta_bar/view.coffee:21]: 👏

File: analytics/articles.js [hunk 8]
```diff
@@ -72,6 +72,14 @@ if(location.pathname.match('/article/')){
...
+      article_id: options.articleId,
+      context_type: options.type,
```
Comment (kanaabe) [inline analytics/articles.js]: Based on what you have now, the `context_type = "article_fixed"`, based on the `getType()` function, but we would probably want the type to be `article_popup` instead. This [doc](https://docs.google.com/spreadsheets/d/17oqFli15x1SmNBu_-fPZhFifMw2BNSPJe9VdOgs1aKs/edit#gid=1499052602) is what we use to stay aligned with the Analytics team on naming. Sorry I hadn't shared this with you before! 

```
context_type: 'article_popup'
impression_type: 'newsletter_signup'
```

File: apps/articles/client/editorial_signup.coffee [hunk 9]
```diff
@@ -39,7 +43,23 @@ module.exports = class EditorialSignupView extends Backbone.View
...
+    @ctaBarView = new CTABarView
+      mode: 'editorial-signup'
+      name: 'dismissed-editorial-signup'
+      persist: true
+      email: sd.CURRENT_USER?.email or ''
+      expires: 2592000
+    if not @ctaBarView.previouslyDismissed() and
+      @canViewCTAPopup() and
+      @eligibleToSignUp() and
+      qs.parse(location.search.replace(/^\?/, '')).utm_source isnt 'sailthru'
```
Comment (kanaabe) [inline apps/articles/client/editorial_signup.coffee]: We should write a test for this if statement :)

**Comment by kanaabe** [comment]
Overall great job on this! There are a couple comments we should address before merging but all minor. 👍

File: analytics/articles.js (hunk 1 above)
Comment (eessex) [reply analytics/articles.js]: I believe these are now in the style, please let me know if anything looks amiss!

**Comment by eessex** [comment]
Thanks for your feedback, looking much cleaner now :)

**Comment by kanaabe** [comment]
Uh oh, I think this needs a rebase. I recently merged a shrinkwrap, so we have a merge conflict now :(

**Comment by eessex** [comment]
@kanaabe just addressed the conflicts in shrinkwrap, should be good to go now.

**Comment by kanaabe** [comment]
🌟 Looks great. Thanks!

**Pull request merged**